lib/collection/src/shards/local_shard/clock_map.rs

```diff
@@ -300,15 +300,16 @@ impl RecoveryPoint {
     pub fn remove_clock_if_newer_or_equal_to_tag(&mut self, tag: ClockTag) -> bool {
         let key = Key::from_tag(tag);
 
-        let mut any_equal = false;
+        let mut is_equal = false;
+
         if let Some(&(tick, _)) = self.clocks.get(&key) {
             if tick >= tag.clock_tick {
                 self.clocks.remove(&key);
+                is_equal = tick == tag.clock_tick;
             }
-            any_equal |= tick == tag.clock_tick;
         }
 
-        any_equal
+        is_equal
     }
```

Comment:
We can also do `return tick == tag.clock_tick;` and just `false` at the end.

I don't have a strong preference for either.